### ISSUE TITLE
Preserve DOCTYPE declarations

### ISSUE THREAD

**Issue by infotexture** [prettier/plugin-xml#5]
This plug-in currently preserves the optional [XML declaration](https://www.w3.org/TR/xml/#NT-XMLDecl) on the first line of XML files:

```xml
<?xml version="1.0" encoding="UTF-8"?>
```

but strips any subsequent [DOCTYPE declarations](https://www.w3.org/TR/xml/#dt-doctype) that are required for DTD-based validation in XML dialects, such as [DocBook](https://docbook.org) or [DITA](http://docs.oasis-open.org/dita/dita/v1.3/dita-v1.3-part0-overview.html).

These declarations should be preserved to allow editors and parsers to validate the file content.

## DocBook example

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE article PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN"
                         "http://docbook.org/xml/4.5/docbookx.dtd">
<article>
    <title/>
    <sect1>
        <title/>
        <para/>
    </sect1>
</article>
```

## DITA example

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE task PUBLIC "-//OASIS//DTD DITA Task//EN" "task.dtd">
<task id="ID">
    <title></title>
    <shortdesc></shortdesc>
    <taskbody>
        <context>
            <p></p>
        </context>
        <steps>
            <step>
                <cmd></cmd>
            </step>
        </steps>
    </taskbody>
</task>
```

**Comment by hermitmaster** [comment]
This is a major issue with hibernate mappings and the config xml in java projects. A solution would be greatly appreciated.

**Comment by kddeisz** [comment]
At the moment I can't get them out of the parser, but I'm working on it here: https://github.com/SAP/xml-tools/issues/81

**Comment by kddeisz** [comment]
Fixed by #43, will be released in upcoming v0.5.0
